cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseConsumerStateStore.java

```diff
@@ -316,6 +328,15 @@ public void configureGroups(Iterable<? extends ConsumerGroupConfig> groupConfigs
     return scanBarriers(groupId, new AllCollector<QueueBarrier>()).finish(new ArrayList<QueueBarrier>());
   }
 
+  /**
+   * Gets all barrier information for all groups. The information are sorted in the order of
+   * the barrier changes.
+   */
+  Multimap<Long, QueueBarrier> getAllBarriers() {
+    return scanBarriers(new AllPairCollector<Long, QueueBarrier>()).finishMultimap(
+      HashMultimap.<Long, QueueBarrier>create());
```

Comment:
Is `HashMutlimap` guarantee sorting order? Shouldn't you use `LinkedHashMultimap`?